### DISCUSSION TITLE
Get Function Level Metrics

### DISCUSSION THREAD

**Discussion by haiakablan** [tomkerkhove/promitor#2157] [Q&A]
Hi, 
I want to clarify if we can get Function level metrics. 
For now, I can only get Function-app level metrics and I need the functions under that function-app.

Thanks, 
Haia

**Comment by tomkerkhove** [comment]
Hi,

Can you share an example of a function level metric please?

**Reply by haiakablan** [reply]
For example, the HTTP5xx metric which is HTTP Server Errors.
actually, can we achieve with promitor the same that we have with azure monitoring, 
as mentioned here  https://learn.microsoft.com/en-us/azure/azure-functions/monitor-functions?tabs=azurecli  ?

**Reply by tomkerkhove** [reply]
Hm, it should be supported - https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported#microsoftwebsites

Did you try this?

**Reply by haiakablan** [reply]
yes, I tried it,  this gave data only for the function app level and not the functions level

**Reply by tomkerkhove** [reply]
What is the outcome if you configure Http5xx as metric?

**Reply by haiakablan** [reply]
![image](https://user-images.githubusercontent.com/116159904/196708640-cd159d9e-9843-4289-a6e6-1224ee8287dd.png)

after Microsoft.Web/sites/ there's the function name

**Reply by tomkerkhove** [reply]
What does the metrics explorer in the portal say? Because it's not really throwing exceptions but merely reporting 0

**Reply by haiakablan** [reply]
these values are for the 9 resources that were discovered -  which are the function-apps and not the functions themselves.   ( see the first line in prev screenshot). 
in this case, the values of the metrics explorer in the portal are 0. 
but if you see there's a "ResourceNotFound" Error - which is related to the function 

Take this example, when using the metric "Http2xx"
![image](https://user-images.githubusercontent.com/116159904/196868689-3fe070f0-f663-400f-bb3f-32da435457b1.png)
I got non-zero values, but again I have that error.

**Reply by tomkerkhove** [reply]
So pulling function-level metrics does work is what I'm seeing :)

Most probably, however, you have an Azure resource which is not accessible to get the metrics for it and we get a 404 on it due to the permissions.

Can you verify the permissions for the scraper is the same on that resource as others?

**Reply by haiakablan** [reply]
ok got it, will check this 
but another question, 
I found only this configuration which is related to the function app https://docs.promitor.io/v2.8/scraping/providers/function-app/
I didn't find something specific for functions, can you advise? and provide the related/sample configuration where you can see it is working

**Reply by tomkerkhove** [reply]
What are you specifically looking for because it links to the available Azure Monitor metrics that it provides?

Functions are part of a Function App so all the metrics are bundled from Azure Monitor perspective

**Reply by haiakablan** [reply]
Is there any way we can check the same metrics on the function level?

For example, if there are 5 times http2xx responses, so to get those details?

**Reply by tomkerkhove** [reply]
I think I'm just not getting it :D Isn't that metric already doing that?
